starter-code/src/components/DisplayContacts.jsx

```diff
@@ -5,12 +5,20 @@ export default class DisplayContacts extends Component {
 
     state = {
         contacts: this.props.contacts,
-        fiveFirstContacts: this.props.contacts.splice(0,5)
+        displayedContacts: this.props.contacts.splice(0,5),
     };
+
+    addContact = () => {
+        var newContact = this.state.contacts[Math.floor(Math.random() * (this.props.contacts.length-5))+5]
+        const copy = [...this.state.displayedContacts];
+        copy.push(newContact);
+        this.setState({displayedContacts:copy});
```

Comment:
Here you could check if the contact doesnt already exist in the displayed contacts otherwise you would get duplication.